src/components/ActivitiesContainer.js

```diff
@@ -0,0 +1,150 @@
+import { createContext, useContext, useEffect, useReducer, useState } from "react";
+import { AuthContext } from "./global/Auth";
+
+export const ActivitiesContext = createContext({
+    activities: [],
+    scheduledActivities: [],
+    unscheduledActivities: [],
+});
+
+export const ActivitiesDispatchContext = createContext(null);
+
+let lastUpdate = new Date().getTime() - 15000;
+
+const ActivitiesReducer = (state, action) => {
+    switch (action.type) {
+        case 'delete' : {
+            if (!action.id) {
+                throw new Error('No id provided to delete activity from state');
+            }
+            const newState = state.activities.reduce(
```

Comment:
Typescript would  not allow a reduce like this. All of the suggested alternatives involved looping through `activities` more than once or use side-effects. This killed typescript for me if, in order to use the tool, I have to make my code less performant.